Java build flakes: "Memory constraints are impeding performance"

Java Wordcount Direct Runner (ubuntu-latest) failed a couple times on me with the following warning/error. It doesn't seem like "impeding performance" should cause the build to fail altogether, so I'm wondering if there's some way we can silence this warning.

```

> Task :sdks:java:extensions:protobuf:compileJava
warning: Memory constraints are impeding performance;
please increase max heap size.

...
> Task :sdks:java:extensions:protobuf:compileJava
error: warnings
found and -Werror specified
Note: Some input files use or override a deprecated API.
Note: Recompile
with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note:
Recompile with -Xlint:unchecked for details.
1 error
1 warning

> Task :sdks:java:extensions:protobuf:compileJava
FAILED
> Task :runners:core-construction-java:compileJava
Note: Some input files use or override a
deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked
or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

FAILURE: Build failed with
an exception.

* What went wrong:
Execution failed for task ':sdks:java:extensions:protobuf:compileJava'.
>
Compilation failed; see the compiler error output for details.

```


Imported from Jira [BEAM-11837](https://issues.apache.org/jira/browse/BEAM-11837). Original Jira may contain additional context.
Reported by: ibzib.

--- a/.test-infra/jenkins/CommonJobProperties.groovy
+++ b/.test-infra/jenkins/CommonJobProperties.groovy
@@ -169,7 +169,7 @@ class CommonJobProperties {
   }
 
   // Default maxWorkers is 12 to avoid jvm oom as in [BEAM-4847].
-  static void setGradleSwitches(context, maxWorkers = 12) {
+  static void setGradleSwitches(context, maxWorkers = 8) {
     def defaultSwitches = [
       // Continue the build even if there is a failure to show as many potential failures as possible.
       '--continue',
@@ -182,9 +182,10 @@ class CommonJobProperties {
 
     // Ensure that parallel workers don't exceed total available memory.
 
-    // For [BEAM-4847], hardcode Xms and Xmx to reasonable values (2g/4g).
+    // Workers are n1-highmem-16 with 104GB
+    // 2 Jenkins executors * 8 Gradle workers * 6GB = 96GB
     context.switches("-Dorg.gradle.jvmargs=-Xms2g")
-    context.switches("-Dorg.gradle.jvmargs=-Xmx4g")
+    context.switches("-Dorg.gradle.jvmargs=-Xmx6g")
 
     // Disable file system watching for CI builds
     // Builds are performed on a clean clone and files aren't modified, so

